### ISSUE TITLE
KoboldCPP Settings not saving? 

### ISSUE THREAD

**Issue by ElfrinoGem** [LostRuins/koboldcpp#876]
**_IMPORTANT NOTE: It's not Cuda12 at all, it's something wonky with my Kobold settings, they don't save!_**

**update:** 

Hey, I found something super strange. I don't think it's Cuda at all but something weird with my KoboldCPP ( I don't know if this is effecting others or just me though).

Each time I load Kobold I need to manually choose a preset and then configure the settings to how I want them. Kobold won't recognize my saved settings even though they 'seem' to be set correctly when i load in. Basically for kobold to see my settings I have to manually adjust them each time I enter the program.

**Comment by LostRuins** [comment]
I doubt there should be much difference. Are you sure your generation settings are the same? Try using MMQ, and disable Flash Attention on both. You can also try using a fixed seed.

**Comment by ElfrinoGem** [comment]
I had been using MMQ but I'll try disabling Flash Attention and trying a fixed seed and see how it goes. Thankyou for the tips.

**Comment by ElfrinoGem** [comment]
After running more tests (trying MMQ and disabling Flash for both versions) the output is still far superior in the cu11 version to the cu12 version. Here is a sample output between the two versions:


**Kobold (cu11)**

_The Marquis Francou Debonali IV had just awoken from a dream, or perhaps it was a vision. He lay in a silk-sheeted bed within the opulence of his palatial estate, high above the shimmering streets of Varisium, a great metropolis that stretched across the shores of an enormous inland sea. The sun, rising beyond the sapphire horizon, cast an otherworldly glow through stained glass windows depicting mythical scenes of long-forgotten legends.

Francou's eyes, an icy shade of blue, wandered the intricate patterns of the room's marble walls. He stroked his neatly trimmed beard, his fingers adorned with rings set with gems that glowed with an internal light. The dream, as was often the case, involved a mysterious woman, her face obscured, a figure that eluded him, flitting through mists that never quite cleared. He sighed, throwing back the heavy brocade curtains to reveal the vastness of the city, its spires and towers reaching toward the sky.

In the distance, the great engine of the Aerium elevated above the rooftops, a testament to the city's might and the advanced technology of that age. The buzzing of the electromagnets could be heard even at this height, a constant reminder of the civilization's achievement. Francou's eyes drifted to the small device on his bedside table—a contraption of cogs and gears that whirred softly, emitting a soft green glow. It was a gambling machine, a recent acquisition, and one that had brought him both great wins and devastating losses.

Today, he decided, would be a day of victory. Dressing swiftly in a robe of deep purple silk, he descended the spiral staircase into his private sanctuary, a museum of curios and relics from a hundred different worlds. The air smelled of exotic spices and ancient tomes; the musty aroma of knowledge. He strolled past a glass case containing an actualized dream-recorder, a device capable of extracting and preserving dreams, capturing them like fireflies in a jar.

His servants, automated droids in the guise of humans, glided soundlessly through the halls, their metallic frames polished to a mirror shine. Francou waved absentmindedly as they scurried about their duties, their mechanical eyes flicking up to acknowledge their master. Passing through a conservatory filled with alien flora, he stepped out onto a balcony that overlooked the sea. The cool air nipped at his skin, and the vast wet plain was dotted with strange creatures that spouted ink-black smoke rings. boats with shimmering hulls skimmed the wave tops, their engines singing a deep song._

**Kobold (cu12)**

_The Marquis Francou Debonali IV, known across the twelve continents as a hedonist and bon vivant, found himself bored one evening, a feeling as unfamiliar as it was unwelcome. He had exhausted the usual pursuits—the thrilling races on the speedscape circuits, the decadent feasts that lasted for days, and even the exotic pleasures of the Pleasure Hive left him unmoved. His closets runneth over with the finest attire, and the staterooms of his palace bulged with the latest gadgets and baubles. What new diversion could stir his jaded soul?

As he strolled the gardens of his palatial estate, the Marquis spied a curious contraption, a recently delivered gadget he had not yet unboxed. The sleek, black oval shape resembling a giant egg bespoke of enigmatic origins. Ancient texts spoke of such devices, rumors of an extraterrestrial kind. With a playful glint in his eye, he activated the mysterious artifact.

A soft hum filled the air, accompanied by a gentle glow emanating from within. The Marquis felt a peculiar sensation, as if his very being were dissolving, atom by atom. Alarm edged into his thoughts, quickly replaced by a strange exhilaration. The darkness pulled him in, engulfing his senses, and he found himself hurtling through an endless tunnel, a cosmic voyage beyond any he had ever experienced.

When the Marquis returned, reformed and refreshed, he found his world changed. The once familiar landscape was distorted, the very laws of physics askew. The sky above was now a kaleidoscope of neon hues, swirling in impossibly shaped clouds. The sun, a distant memory, was replaced by an eerie luminescence that cast an ethereal glow upon the transformed Earth.

He soon discovered that time itself had warped, each day blending into the next with no discernible pattern. Seasons were but fleeting moments, or so it seemed to the bemused Marquis, as entire years raced by in a matter of hours. Trees grew to their full bloom in minutes, only to wither and die in the blink of an eye._

**Comment by ElfrinoGem** [comment]
Hey, I found something super strange. I don't think it's Cuda at all but something weird with my KoboldCPP ( I don't know if this is effecting others or just me though).

Each time I load Kobold I need to manually choose a preset and then configure the settings to how I want them. Kobold won't recognize my saved settings even though they 'seem' to be set correctly when i load in. Basically for kobold to see my settings I have to manually adjust them each time I enter the program.

**Comment by LostRuins** [comment]
Settings are saved in the JSON file (or autosave, if you dont load from file). are you perhaps overwriting them with a different JSON?

**Comment by ElfrinoGem** [comment]
Ahh, I have a whole bunch of Kobold versions in the same directory and I use them interchangeably. Do you think they could be interfering with each other? Where is the JSON file stored?

**Comment by LostRuins** [comment]
Are you using the built-in save slots?

**Comment by ElfrinoGem** [comment]
Yes I use them constantly to save my stories. Ahh, got it, settings are saved with the story. Thankyou.  Is it possible to save your own settings preset? (aka Mayday, Goodwinds, etc) so it appears in the drop down window?

**Comment by LostRuins** [comment]
Nope, the presets are fixed. However, you can easily save your desired generation settings along with each story.

You can also selectively choose what to load by enabling "Advanced Load" in settings.

**Comment by ElfrinoGem** [comment]
Thankyou.
